src/App.vue

```diff
@@ -1,28 +1,20 @@
 <template>
   <div id="app">
     <img alt="Vue logo" src="./assets/logo.png" />
-    <HelloWorld msg="Hello Vue 2 + Vite" />
+    <hello-world-component msg="Hello Vue 2 + Vite" />
 
     <h1>App Component: {{ applicationNumber }}</h1>
     <button @click="applicationNumber+=10">Increment Ten From Parent Cmponent</button>
-    <Counter v-model="applicationNumber" @increment="applicationNumber++" @decrement="applicationNumber--" />
+    <counter-component v-model="applicationNumber" />
   </div>
 </template>
 
-<script>
-import HelloWorld from './components/HelloWorld.vue';
-import Counter from './components/Counter.vue';
-export default {
-  components: {
-    HelloWorld,
-    Counter
-  },
-  data() {
-    return {
-      applicationNumber: 10
-    }
-  }
-};
+<script setup>
+import { ref } from '@vue/composition-api'
+import HelloWorldComponent from './components/HelloWorld.vue';
+import CounterComponent from './components/Counter.vue';
```

Comment:
@DaudIbrahim A minor naming convention: Component names shouldn't be suffixed by 'component', as it's self explanatory that these are components.